File: zmin9/BOJ_5585번_거스름돈_Greedy.js
const fs = require('fs');

// 입력 처리
//const N = parseInt(fs.readFileSync('/dev/stdin').toString());
const N = parseInt(fs.readFileSync('./input.txt').toString());

// 구해야하는 것이 '잔돈 개수'임에 유의할 것
let change = 1000 - N;
let result = 0;

// 가지고 있는 잔돈 종류
const coins = [500, 100, 50, 10, 5, 1];

// 큰 단위의 잔돈을 많이 쓸 수록 전체 잔돈의 개수가 적어지므로
// 가장 큰 단위의 잔돈부터 거슬러줄 수있는 최대값을 거슬러준다
coins.forEach((coin)=>{
    if(N<=0) return;
    result = result + Math.floor(change / coin);
    change %= coin;
});

console.log(result);

File: jieun^,^/BOJ_5585번_거스름돈_Greedy.py
# 잔돈 - 500, 100, 50, 10, 5, 1
# 거스름돈의 개수가 가장 적게
# 잔돈은 항상 충분하게 준비 되어 있음

# 항상 1000을 냄
# 입력 price - 물건 값 (1<=price<1000)
price = int(input())

# 출력 ans - 잔돈의 개수
ans = 0

change = 1000 - price

# ex) 620 -> 500x1 + 100x1 + 10x2
# 최대한 큰 돈의 단위로 거스름돈을 지불해야 개수가 최소
money = [500, 100, 50, 10, 5, 1]
for m in money:
    x = change//m
    if x>=1:
        ans += x
        change %=m
    
print(ans)